c++/src/CMakeLists.txt

```diff
@@ -125,6 +125,83 @@ include_directories (
   ${LZ4_INCLUDE_DIRS}
   )
 
+# To avoid reading the Timezone database from disk, we load the file during the
+# build and inject them as binary into the library. This can increase the size
+# of library but avoids doing system calls at runtime. The behavior can be
+# disabled by using the CMake flag NO_EMBEDDED_TZ_DB
+if (NOT DEFINED EMBEDDED_TZ_DB)
+  set(EMBEDDED_TZ_DB ON)
+endif()
+if (EMBEDDED_TZ_DB)
+  message(STATUS "Caching of timezone data enabled.")
+  set(DEFAULT_TIMEZONE_DIR "/usr/share/zoneinfo")
+
+  # Check if zone1970.tab is present, otherwise select zone.tab
+  # Parse zone tab to extract all availble zones
+  execute_process(COMMAND find ${DEFAULT_TIMEZONE_DIR} -type f
+    COMMAND grep -v "posix"
+    COMMAND grep -v "right"
+    COMMAND grep -v "tab"
+    RESULT_VARIABLE TZ_LIST_RC
+    OUTPUT_VARIABLE TZ_LIST_OUTPUT
+    ERROR_VARIABLE TZ_ERROR)
+
+  if (TZ_LIST_RC)
+    message(FATAL_ERROR "Could not parse Timezone database.")
+  endif()
+
+  set(FILE2ARRAY "${CMAKE_CURRENT_SOURCE_DIR}/../file2array.sh")
+
+  # For each zone generate a target
+  # Add each target to dependency on Timezone.cc
+  set(TZ_DEPS "")
```

Comment:
hoist this initialization out of the `if (EMBEDDED_TZ_DB)`?
That way we can safely add it as a dependency to orc library on line 214